### ISSUE TITLE
Add SWA support to Pytorch trainer

### ISSUE THREAD

**Issue by andrewyates** [capreolus-ir/capreolus#88]
Pytorch 1.6 supports [stochastic weight averaging (SWA)](https://pytorch.org/blog/pytorch-1.6-now-includes-stochastic-weight-averaging/), which could improve model effectiveness. The Pytorch trainer should support a boolean `swa` config option that controls whether it is used.

As described in the docs, using SWA with batchnorm requires additional steps. Can we automatically detect whether a model contains any batchnorm layers and, if so, either raise an exception (easier) or perform the necessary additional steps?
